packages/devtools_app/lib/src/vm_developer/vm_statistics_view.dart

```diff
@@ -0,0 +1,304 @@
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../common_widgets.dart';
+import '../table.dart';
+import '../table_data.dart';
+import '../utils.dart';
+import 'vm_developer_common_widgets.dart';
+import 'vm_developer_tools_screen.dart';
+import 'vm_service_private_extensions.dart';
+import 'vm_statistics_view_controller.dart';
+
+class VMStatisticsView extends VMDeveloperView {
+  const VMStatisticsView()
+      : super(
+          id,
+          title: 'VM',
+          icon: Icons.devices_sharp,
+        );
+  static const id = 'vm-statistics';
+
+  @override
+  Widget build(BuildContext context) => VMStatisticsViewBody();
+}
+
+/// Displays general information about the state of the Dart VM.
+class VMStatisticsViewBody extends StatelessWidget {
+  final controller = VMStatisticsViewController();
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: controller.refreshing,
+      builder: (context, refreshing, _) {
+        return Column(
+          children: [
+            Row(
+              children: [
+                RefreshButton(
+                  onPressed: controller.refresh,
+                ),
+              ],
+            ),
+            Expanded(
+              child: VMStatisticsWidget(
+                controller: controller,
+              ),
+            ),
+          ],
+        );
+      },
+    );
+  }
+}
+
+class VMStatisticsWidget extends StatelessWidget {
+  const VMStatisticsWidget({@required this.controller});
+
+  final VMStatisticsViewController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Flexible(
+          flex: 3,
+          child: Column(
+            children: [
+              GeneralVMStatisticsWidget(
+                controller: controller,
+              ),
+              ProcessStatisticsWidget(
+                controller: controller,
+              )
+            ],
+          ),
+        ),
+        Flexible(
+          flex: 4,
+          child: Column(
+            children: [
+              IsolatesPreviewWidget(
+                controller: controller,
+              ),
+              IsolatesPreviewWidget(
+                controller: controller,
+                systemIsolates: true,
+              ),
+            ],
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+/// Displays general information about the VM instance, including:
+///   - VM debug name
+///   - Dart version
+///   - Embedder name
+///   - Start time
+///   - Profiler mode
+///   - Current memory consumption
+class GeneralVMStatisticsWidget extends StatelessWidget {
+  const GeneralVMStatisticsWidget({@required this.controller});
+
+  final VMStatisticsViewController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    final vm = controller.vm;
+    return Flexible(
```

Comment:
same comment here about moving Flexible to wrap GeneralVMStatisticsWidget where it is declared instead of here